flink-libraries/flink-cep/src/main/java/org/apache/flink/cep/pattern/conditions/RichNotCondition.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.cep.pattern.conditions;
+
+import org.apache.flink.api.common.functions.RuntimeContext;
+import org.apache.flink.api.common.functions.util.FunctionUtils;
+import org.apache.flink.configuration.Configuration;
+
+/**
+ * A {@link RichIterativeCondition condition} which negates the condition it wraps
+ * and returns {@code true} if the original condition returns {@code false}.
+ *
+ * @param <T> Type of the element to filter
+ */
+public class RichNotCondition<T> extends RichIterativeCondition<T> {
+
+	private static final long serialVersionUID = 1L;
+
+	private final IterativeCondition<T> original;
+
+	public RichNotCondition(final IterativeCondition<T> original) {
+		this.original = original;
```

Comment:
I think we should unify the default behavior for null in `RichOr` ， `RichAnd` and `RichNot` What do you think? @dianfu @dawidwys

Reply:
I think this is by design as CEP support patterns where there are no conditions. You can refer to the test case of NFAITCase#testNoConditionNFA.

Reply:
+1

Reply:
Yes, I see that case, but why can't we let `RichAnd` and `RichOr` have default handling for null?

Reply:
Sounds good！